File: core/NAryTreePostorderTraversal/NAryTreePostorderTraversal.py
from typing import List


class Node:
    def __init__(self, val=None, children=None):
        self.val = val
        self.children = children


def postorder(root: 'Node') -> List[int]:
    if root == None:
        return []

    res = []
    for node in root.children:
        res.extend(postorder(node))
    res.append(root.val)
    return res

File: core/NAryTreePostorderTraversal/NAryTreePostorderTraversal.java
package com.dbc;

import java.util.ArrayList;
import java.util.List;


public class NAryTreePostorderTraversal {
    public class Node {
        public int val;
        public List<Node> children;

        public Node() {
        }

        public Node(int _val) {
            val = _val;
        }

        public Node(int _val, List<Node> _children) {
            val = _val;
            children = _children;
        }
    }

    public List<Integer> postorder(Node root) {
        List<Integer> res = new ArrayList<>();
        if (root == null) {
            return res;
        }
        for (Node node : root.children) {
            res.addAll(postorder(node));
        }
        res.add(root.val);
        return res;
    }
}
